pipeline/src/pipeline-config/pipelineDatabase.ts

```diff
@@ -0,0 +1,181 @@
+import { PostgresClient } from '@jvalue/node-dry-pg'
+import { ClientBase, PoolConfig, QueryResult } from 'pg'
+
+import { POSTGRES_DB, POSTGRES_HOST, POSTGRES_PORT, POSTGRES_PW, POSTGRES_SCHEMA, POSTGRES_USER } from '../env'
+import { PipelineConfig, PipelineConfigDTO } from './model/pipelineConfig'
+
+const CONFIG_TABLE_NAME = 'PipelineConfigs'
+const OUTBOX_TABLE_NAME = 'outbox'
+
+const CREATE_CONFIG_TABLE_STATEMENT = `
+  CREATE TABLE IF NOT EXISTS "${POSTGRES_SCHEMA}"."${CONFIG_TABLE_NAME}" (
+  "id" bigint NOT NULL GENERATED ALWAYS AS IDENTITY,
+  "datasourceId" bigint NOT NULL,
+  "func" varchar NOT NULL,
+  "author" varchar,
+  "displayName" varchar NOT NULL,
+  "license" varchar,
+  "description" varchar,
+  "createdAt" timestamp,
+  CONSTRAINT "Data_pk_${POSTGRES_SCHEMA}_${CONFIG_TABLE_NAME}" PRIMARY KEY (id)
+)`
+const INSERT_CONFIG_STATEMENT = `
+  INSERT INTO "${POSTGRES_SCHEMA}"."${CONFIG_TABLE_NAME}"
+  ("datasourceId", "func", "author", "displayName", "license", "description", "createdAt")
+  VALUES ($1, $2, $3, $4, $5, $6, $7)
+  RETURNING *`
+const UPDATE_CONFIG_STATEMENT = `
+  UPDATE "${POSTGRES_SCHEMA}"."${CONFIG_TABLE_NAME}"
+  SET "datasourceId"=$2, "func"=$3, "author"=$4, "displayName"=$5, "license"=$6, "description"=$7
+  WHERE id=$1`
+const GET_CONFIG_STATEMENT = `
+  SELECT * FROM "${POSTGRES_SCHEMA}"."${CONFIG_TABLE_NAME}" WHERE "id" = $1`
+const GET_ALL_CONFIGS_STATEMENT = `
+  SELECT * FROM "${POSTGRES_SCHEMA}"."${CONFIG_TABLE_NAME}"`
+const GET_ALL_CONFIGS_BY_DATASOURCE_ID_STATEMENT = `
+  SELECT * FROM "${POSTGRES_SCHEMA}"."${CONFIG_TABLE_NAME}" WHERE "datasourceId" = $1`
+const DELETE_CONFIG_STATEMENT = `
+  DELETE FROM "${POSTGRES_SCHEMA}"."${CONFIG_TABLE_NAME}" WHERE "id" = $1 RETURNING *`
+const DELETE_ALL_CONFIGS_STATEMENT = `
+  DELETE FROM "${POSTGRES_SCHEMA}"."${CONFIG_TABLE_NAME}" RETURNING *`
+
+const CREATE_OUTBOX_TABLE_STATEMENT = `
```

Comment:
Does it make sense to separate pipeline handling from pipelineEvent handling?

Reply:
I have thought of this, but did not do it because the initialization (e.g. creating the database tables) should be done in the same transaction in the [init function](https://github.com/jvalue/open-data-service/pull/311/files/9df84569714ae1e4ff32990fce656c53faf3242b#diff-7344c05a5acea5f112528cd03003ce272755d6f25349df9d2e898a57beaff7e1R78-R86).

Reply:
After thinking more about this, we could actually do this by splitting it into three parts: The init function, the pipeline config handling and the pipeline event handling. The init function would then just call a create table functions for the pipeline config and the pipeline event table

Reply:
I have just pushed a new commit containing the proposed split. It is now a much better separation of concerns 🚀